src/bootstrap/builder.rs

```diff
@@ -850,7 +850,42 @@ impl<'a> Builder<'a> {
                 cargo.args(s.split_whitespace());
             }
             rustflags.env("RUSTFLAGS_BOOTSTRAP");
-            rustflags.arg("--cfg=bootstrap");
+            if cmd == "clippy" {
+                // clippy overwrites any sysroot we pass on the command line.
```

Comment:
I am confused, this sysroot *is* being passed on the command line? If that gets overwritten, then what's the point?

Reply:
The comment is wrong - it gets overwritten if we pass it to _cargo_, not if we pass it to clippy.

https://github.com/rust-lang/rust-clippy/blob/a771557ee92af5e313714f127bb48a1d787a1cff/src/driver.rs#L310